api/capi/include/nntrainer.h

```diff
@@ -215,6 +215,49 @@ int ml_train_model_set_optimizer(ml_train_model_h model,
 int ml_train_model_set_dataset(ml_train_model_h model,
                                ml_train_dataset_h dataset);
 
+/**
+ * @brief Save the model after training
+ * @details Use this function to save the current model.
+ * When calling with @a ML_TRAIN_MODEL_SAVE_LOAD_FLAGS_INFERENCE_PARAMS or
+ * ML_TRAIN_MODEL_SAVE_LOAD_FLAGS_TRAINING_PARAMS, the model must be compiled
+ * with @a ml_train_model_compile beforehand.
+ *
+ * @since_tizen 6.5
+ * @param[in] model The NNTrainer model handle to save
+ * @param[in] path_prefix  Path prefix to save the file. This function will save
+ * one or number of files using the given prefix. Note that this does not state
+ * the exact file name but prefix
+ * @param[in] option Option flag which part of model should be saved
```

Comment:
Usually, when we are talking about saving the model, I think it might be better to make sure not to overwrite the current model or limit the number of saved models and kind of naming rule. I'm sure you have a plan for this.

Reply:
Talked offline to clarify, that we will have training params including inference params :)

Reply:
So, what is the final conclusion on the overwrite/numbered saves?

Reply:
1. If we number with given path,

- Over 100mb storage will be implicitly used everytime calling this api.
- It's hard to locate which to delete. Because the name mangling rule won't be exposed.
- So, user can only delete the whole model.

2. If we override,

- saving on the same path with different flag is grey area in terms of api behavior.
 eg) saving train params > saving inference params > saving train params (in *.bin file, this overrides inference params), (in otherfiles this might not override inference params)

3. If give error if the path is taken.

- Not user friendly if user wants to override.
- But predictable, and user can control over the behavior if they wants to.

IMO, the option 3 has the only way to guarantee the consistent semantics is to give error if the given path is taken.

side note) We do not support deletion of the saved model. With path_prefix strategy, it's not possible to easily delete a file, so I think it's better to get empty directory as an argument instead. 
Come to think of it, it seems we should have a directory as an argument instead of a path prefix.